eth/db/chain.py

```diff
@@ -93,15 +94,18 @@ def get_block_uncles(self, uncles_hash: Hash32) -> Tuple[BlockHeaderAPI, ...]:
     def _set_as_canonical_chain_head(
             cls,
             db: DatabaseAPI,
-            block_hash: Hash32,
+            header: BlockHeaderAPI,
```

Comment:
Splitting the PR apart a bit would really help. Like this change looks interesting: I can give it a better review in a self-contained PR.

Reply:
Yeah, makes sense!

Reply:
https://github.com/ethereum/py-evm/pull/1891